### ISSUE TITLE
библиотека обертка над rac/ras

### ISSUE THREAD

**Issue by ghost** [EvilBeaver/oscript-library#40]
Зафиксирую - чтобы осталось

Нужно управлять rac/ras - то есть управлять сервером приложений НЕ через COM

Пока не знаю как сделать и кто будет, но пусть останется

Имеется ввиду вот такое http://oparin.info/1c/administrirovanie-serverov-1s-pod-linux-ubuntu

**Comment by nixel2007** [comment]
Поддерживаю! У самого было желание что-то такое накидать. Для rac/ras ещё очень полезной фичей является взаимодействие через веб-сервер по обычным http-запросам.

**Comment by EvilBeaver** [comment]
В принципе,есть в БСП, но в минимальном объеме. А еще есть JAVA библиотека+[IKVM](http://www.ikvm.net/)

**Comment by ghost** [comment]
@EvilBeaver Java - ни коем случае. Если коротко - то оно НЕ работает о слова совсем. 

Длинно могу сформулировать попозже - там 1С-ники "молодцы" причем "конкретные молодцы"
Вот тут немного описано http://1c-dn.com/forum/forum17/topic882/?sphrase_id=39840#message3402

**Comment by ghost** [comment]
@EvilBeaver получить правильные библиотеки Java можно только если купить лицензию с 1C-dn сайта. Русские акаунты не работают - текущее Java API опубликованное на сайте ИТС устарели и не обновляются. а на 1С-DN обновляются вручную ;-), как показала практика, но купить на 1C-DN можно только не из России, точнее если сделать фейковый акаунт "из-за пределов СНГ"

**Comment by EvilBeaver** [comment]
@allustin ну то, что они жопы, мы все как бы знаем) Остается еще надежда на реверсеров уровня awa и orefkov

**Comment by EvilBeaver** [comment]
Да, и еще есть вот такое https://github.com/alekseybochkov/WebRAS
Как концепт мне нравится. Реализация, конечно, из костылей с применением JAVA (@allustin - она похоже как-то работает) .

**Comment by ghost** [comment]
@EvilBeaver так у Алексея доступ то на 1С-DN есть, он же вроде как активно в Сан-Франциско 1С внедряет ;-). Но пока это конечно костыли.

**Comment by ghost** [comment]
пока не забыл

> 
> REST интерфейс администрирования
> 
>  На замену оснастки Microsoft Management Console и кроссплатформенных утилит ras/rac пришел новый компонент - сервис управления кластером. Он предоставляет REST интерфейс для настройки кластера. Все данные передаются в формате JSON, что облегчает интеграцию с любой инфраструктурой, и не ограничивает выбор технологий.
> 
>     Графический интерфейс администрирования в ознакомительной версии отсутствует, но он будет добавлен в одной из следующих версий.

отсюда http://v8.1c.ru/overview/release_8_4_1/

то есть будущая библиотека явно будет работать в двух режимах

**Comment by pumbaEO** [comment]
И как обічно ни лицензии не будет, ни исходников. И сделано будет для мішколюбов. Не верю я в нормальній интерфейс в библиотеке.

**Comment by EvilBeaver** [comment]
мишколюбы - это девочки на 14 февраля?

**Comment by pumbaEO** [comment]
*мышко-любы

**Comment by EvilBeaver** [comment]
Ну по идее, консольный режим для кнопколюбов уже существует, осталось к нему json прикрутить и гуи на джаве.

Т.е. мишколюбы вторым номером идут, а консоль уже поддержана (правда, не шибко юзабельна)
